### PR TITLE
Extend go-cid to implement the ipld-prime Link interface

### PR DESCRIPTION
**Pull request by willscott**
[Ipld-prime](https://github.com/ipld/go-ipld-prime) uses the term `Link` to describe what a CID does, and uses a [new vocabulary](https://github.com/ipld/go-ipld-prime/blob/8fef5312eb1265f44d42ee4cb2d2bca9dc516327/link.go) for talking about how a Cid and Prefix relate. This adds the requisite methods `Prototype` and `BuildLink` to allow the current implementation to fullfill the IPLD-prime interfaces so that existing CIDs can be directly used as links.

### CODE REVIEW

File: cid.go [hunk 1]
```diff
@@ -589,6 +595,13 @@ func (p Prefix) Sum(data []byte) (Cid, error) {
 	}
 }
 
+// BuildLink returns a new Link value based on the given hashsum,
+// and allows Prefix to satisfy the LinkPrototype interface.
+func (p Prefix) BuildLink(hashsum []byte) Cid {
+	c, _ := p.Sum(hashsum)
```
Comment (mvdan) [inline cid.go:601]: panic if there's an error, instead? otherwise latent bugs could go unnoticed

Comment (willscott) [reply cid.go:601]: I think this does what you want, in that if an invalid - (i guess i'm thinking like `[]byte{}` but even that is valid?) sum is given the response is Cid.Undef which can be matched as an error case still

Comment (mvdan) [reply cid.go:601]: Ah, I see. I think that warrants a mention in the godoc, or at least a comment above the line that ignores the error. If I was reading this function declaration alone, I'd have guessed there was a bug :)

Comment (willscott) [reply cid.go:603]: extended comment

**Pull request closed without merging**